MCM stumbles over invalid AMI ID (AWS)

**What happened**:

I tried to create a cluster which did not come up. I suspect the root cause for this is an invalid/deleted ami image on AWS. 

What I saw is that mcm explodes on this - presumable the missing ami image (this is the log output of the mcm):

```
[...]
W0611 10:55:21.476443       1 client_config.go:548] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0611 10:55:21.477765       1 leaderelection.go:187] attempting to acquire leader lease  shoot--core--dm-jeos/machine-controller-manager...
I0611 10:55:21.766983       1 leaderelection.go:196] successfully acquired lease shoot--core--dm-jeos/machine-controller-manager
I0611 10:55:21.767168       1 event.go:221] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"shoot--core--dm-jeos", Name:"machine-controller-manager", UID:"dbaf8217-8c28-11e9-a125-0a580a3009da", APIVersion:"v1", ResourceVersion:"2857289363", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' machine-controller-manager-67fdd6db6-9t82w became leader
I0611 10:55:22.483820       1 controllermanager.go:273] Starting shared informers
I0611 10:55:22.884020       1 controller.go:475] Starting machine-controller-manager
I0611 10:55:22.975325       1 machine.go:339] Creating machine "shoot--core--dm-jeos-worker-tkg6a-z1-6b8695c5b5-thrxr", please wait!
I0611 10:55:22.975325       1 machine.go:339] Creating machine "shoot--core--dm-jeos-worker-tkg6a-z1-6b8695c5b5-zskmj", please wait!
E0611 10:55:23.391236       1 runtime.go:66] Observed a panic: "index out of range" (runtime error: index out of range)
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:44
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_aws.go:74
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:341
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:192
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:108
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:535
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:552
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:514
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 600 [running]:
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1b814e0, 0x3ac71a0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/gardener/machine-controller-manager/pkg/driver.(*AWSDriver).Create(0xc0007fc340, 0x0, 0x1f016b1, 0x21, 0xc001483960, 0x1, 0x1)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_aws.go:74 +0x1779
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).machineCreate(0xc000bb4000, 0xc00062d180, 0x2214b20, 0xc0007fc340, 0x21cdca0, 0xc001492090)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:341 +0xbc
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).reconcileClusterMachine(0xc000bb4000, 0xc00062d180, 0x0, 0x0)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:192 +0x6c2
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).reconcileClusterMachineKey(0xc000bb4000, 0xc000fccaf0, 0x4a, 0x18, 0xc0013f8d10)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:108 +0x23f
github.com/gardener/machine-controller-manager/pkg/controller.worker.func1.1(0x223d960, 0xc0007c7140, 0xc00043c6f0, 0x458701, 0xf, 0x1ee63fd, 0xe, 0x203000)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:535 +0xe3
github.com/gardener/machine-controller-manager/pkg/controller.worker.func1()
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:552 +0x8d
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0007fc040)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007fc040, 0x3b9aca00, 0x0, 0x1, 0x0)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/gardener/machine-controller-manager/pkg/controller.createWorker.func1(0x223d960, 0xc0007c7140, 0x1ee63fd, 0xe, 0xf, 0x1, 0xc00043c6f0, 0x0, 0xc0006f4040)
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:514 +0xa0
created by github.com/gardener/machine-controller-manager/pkg/controller.createWorker
	/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:513 +0xbe
```

**What you expected to happen**:

MCM does not explode but handles the problem with an error message.

**How to reproduce it (as minimally and precisely as possible)**:

There is a```dm-jeos``` cluster in the ```core``` project in canary which currently suffers the problem with the non existing image and he mcm dump.

**Anything else we need to know?**:

The image id will probably get fixed tomorrow.

--- a/pkg/driver/driver_aws.go
+++ b/pkg/driver/driver_aws.go
@@ -70,6 +70,10 @@ func (d *AWSDriver) Create() (string, string, error) {
 	}
 	metrics.APIRequestCount.With(prometheus.Labels{"provider": "aws", "service": "ecs"}).Inc()
 
+	if len(output.Images) < 1 {
+		return "Error", "Error", fmt.Errorf("Image %s not found", *imageID)
+	}
+
 	var blkDeviceMappings []*ec2.BlockDeviceMapping
 	deviceName := output.Images[0].RootDeviceName
 	volumeSize := d.AWSMachineClass.Spec.BlockDevices[0].Ebs.VolumeSize

